Xcode 14 Native Watch App _WatchKitStub/WK could not be copied

<!-- 🚨 STOP 🚨 𝗦𝗧𝗢𝗣 🚨 𝑺𝑻𝑶𝑷 🚨 -->
We have added a native WatchOS APP for our main App in the csproj file via `_ResolvedWatchAppReferences` Build target. This works on Xcode 13 very good, but not on Xcode 14. The native Watch App does not contain the WatchkitStub any more, so copying the native App will fail with the following message:
```
"/Users/sprenger_s/source/bluesecur_test/Hoermann.BleApp.sln" (Hoermann_BleApp_iOS Ziel) (1) ->
"/Users/sprenger_s/source/bluesecur_test/src/Hoermann.BleApp.iOS/Hoermann.BleApp.iOS.csproj" (Standardziel) (2) ->
(_CoreCreateIpa Ziel) -> 
  /Library/Frameworks/Mono.framework/External/xbuild/Xamarin/iOS/Xamarin.iOS.Common.targets(420,3): error MSB3030: Datei "../../native/Hoermann.BleApp.iOS.Native/output/Hoermann/Release/watchos/Hoermann.BleApp.iOS.Native.WatchOS.app/_WatchKitStub/WK" konnte nicht kopiert werden, da die Datei nicht gefunden wurde. [/Users/sprenger_s/source/bluesecur_test/src/Hoermann.BleApp.iOS/Hoermann.BleApp.iOS.csproj]
```

<!--
Please fill in the *entire* template below.
The template is designed to avoid unnecessary delays to confirm and fix reported issues. Issues with missing information generally end up with a `need-info` label and take significantly more time to fix. Please help us help you!
For more information see https://github.com/xamarin/xamarin-macios/wiki/Submitting-Bugs-&-Suggestions
For support requests use https://www.xamarin.com/support
-->
### Steps to Reproduce

1. Create native Watch OS App
2. Dito must copy the App via the `_ResolvedWatchAppReferences`
3. 

### Expected Behavior

### Actual Behavior
Building fails with the following error:
```
"/Users/sprenger_s/source/bluesecur_test/Hoermann.BleApp.sln" (Hoermann_BleApp_iOS Ziel) (1) ->
"/Users/sprenger_s/source/bluesecur_test/src/Hoermann.BleApp.iOS/Hoermann.BleApp.iOS.csproj" (Standardziel) (2) ->
(_CoreCreateIpa Ziel) -> 
  /Library/Frameworks/Mono.framework/External/xbuild/Xamarin/iOS/Xamarin.iOS.Common.targets(420,3): error MSB3030: Datei "../../native/Hoermann.BleApp.iOS.Native/output/Hoermann/Release/watchos/Hoermann.BleApp.iOS.Native.WatchOS.app/_WatchKitStub/WK" konnte nicht kopiert werden, da die Datei nicht gefunden wurde. [/Users/sprenger_s/source/bluesecur_test/src/Hoermann.BleApp.iOS/Hoermann.BleApp.iOS.csproj]
```

### Environment

<!--
1. How to get version information: https://github.com/xamarin/xamarin-macios/wiki/Diagnosis#version-information
2. Paste into the code block below (between ```)
-->

<details>
<summary>Version information</summary>

```
Visual Studio Community 2022 for Mac
Version 17.3.5 (build 0)
Installation UUID: 80f3f977-05b7-4f99-8778-46c76dd0548e

Runtime
.NET 6.0.5 (64-bit)
Architecture: Arm64

Roslyn (Language Service)
4.3.0-3.22312.2+52adfb8b2dc71ed4278debcf13960f2116868608

NuGet
Version: 6.2.1.2

.NET SDK (Arm64)
SDK: Not installed

.NET Runtime (Arm64)
Runtime: /usr/local/share/dotnet/x64/dotnet
Runtime Versions:
	6.0.6
	6.0.5
	6.0.4
	6.0.3
	6.0.2
	6.0.1
	5.0.17 (x64 – Unsupported)
	5.0.16 (x64 – Unsupported)
	5.0.15 (x64 – Unsupported)
	5.0.14 (x64 – Unsupported)
	5.0.13 (x64 – Unsupported)
	3.1.28 (x64 – Unsupported)
	3.1.27 (x64 – Unsupported)
	3.1.26 (x64 – Unsupported)
	3.1.25 (x64 – Unsupported)
	3.1.24 (x64 – Unsupported)
	3.1.23 (x64 – Unsupported)
	3.1.22 (x64 – Unsupported)

Xamarin.Profiler
Version: 1.8.0.19
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

Updater
Version: 11

Xamarin.Android
Version: 13.0.0.0 (Visual Studio Community)
Commit: xamarin-android/d17-3/030cd63
Android SDK: /Users/sprenger_s/Library/Android/sdk
	Supported Android versions:
		12.1 (API level 32)
		12.0 (API level 31)
		11.0 (API level 30)
		10.0 (API level 29)
		9.0  (API level 28)
		13.0 (API level 33)

SDK Command-line Tools Version: 7.0
SDK Platform Tools Version: 33.0.3
SDK Build Tools Version: 32.0.0

Build Information: 
Mono: dffa5ab
Java.Interop: xamarin/java.interop/d17-3@7716ae53
SQLite: xamarin/sqlite/3.38.5@df4deab
Xamarin.Android Tools: xamarin/xamarin-android-tools/main@14076a6

Microsoft Build of OpenJDK
Java SDK: /Library/Java/JavaVirtualMachines/microsoft-11.jdk
11.0.13
Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

Eclipse Temurin JDK
Java SDK: /Library/Java/JavaVirtualMachines/temurin-8.jdk
1.8.0.302
Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

Android SDK Manager
Version: 17.3.0.23
Hash: 965bf40
Branch: remotes/origin/d17-3
Build date: 2022-09-09 03:49:37 UTC

Android Device Manager
Version: 0.0.0.1169
Hash: fafb1d5
Branch: fafb1d5
Build date: 2022-09-09 03:49:37 UTC

Xamarin Designer
Version: 17.3.0.208
Hash: 0de472ea0
Branch: remotes/origin/d17-3
Build date: 2022-09-09 03:49:29 UTC

Apple Developer Tools
Xcode 14.0 (21335)
Build 14A309

Xamarin.Mac
Xamarin.Mac not installed. Can't find /Library/Frameworks/Xamarin.Mac.framework/Versions/Current/Version.

Xamarin.iOS
Version: 15.99.0.464 (Visual Studio Community)
Hash: 9a839d6c4
Branch: xcode14
Build date: 2022-09-13 05:38:29-0400

Build Information
Release ID: 1703050000
Git revision: ff0de72f31c7c2eccb19854fbaf050fa3f713a7c
Build date: 2022-09-09 03:47:01+00
Build branch: release-17.3
Build lane: release-17.3

Operating System
Mac OS X 12.6.0
Darwin 21.6.0 Darwin Kernel Version 21.6.0
    Mon Aug 22 20:19:52 PDT 2022
    root:xnu-8020.140.49~2/RELEASE_ARM64_T6000 arm64


```

</details>

### Build Logs

<!--
1. How to get a build log: https://github.com/xamarin/xamarin-macios/wiki/Diagnosis#build-logs
2. Place the cursor below this comment block.
3. Drag and drop the build log.
-->

### Example Project (If Possible)

<!--
1. Place cursor below this comment block.
2. Drag and drop the compressed project or files needed to reproduce.
-->
[builds.txt](https://github.com/xamarin/xamarin-macios/files/9651789/builds.txt)

<!--
Switch to the "Preview" tab to ensure your issue renders correctly.
-->

--- a/msbuild/Xamarin.iOS.Tasks/Tasks/ValidateAppBundleTaskBase.cs
+++ b/msbuild/Xamarin.iOS.Tasks/Tasks/ValidateAppBundleTaskBase.cs
@@ -115,11 +115,12 @@ namespace Xamarin.iOS.Tasks {
 				break;
 			}
 		}
-
+		
 		void ValidateWatchApp (string path, string mainBundleIdentifier, string mainShortVersionString, string mainVersion)
 		{
 			var name = Path.GetFileNameWithoutExtension (path);
 			var info = Path.Combine (path, "Info.plist");
+			var isSingleProject = false;
 			if (!File.Exists (info)) {
 				Log.LogError (7014, path, MSBStrings.E7014, name);
 				return;
@@ -162,26 +163,32 @@ namespace Xamarin.iOS.Tasks {
 				Log.LogError (7019, info, MSBStrings.E7019, name, wkCompanionAppBundleIdentifier, mainBundleIdentifier);
 
 			PBoolean watchKitApp;
-			if (!plist.TryGetValue ("WKWatchKitApp", out watchKitApp) || !watchKitApp.Value)
-				Log.LogError (7020, info, MSBStrings.E7020, name);
+			if (plist.TryGetValue ("WKWatchKitApp", out watchKitApp)) {
+				if (!watchKitApp.Value)
+					Log.LogError (7020, info, MSBStrings.E7020, name);
+			} else {
+				isSingleProject = true;
+			}
 
 			if (plist.ContainsKey ("LSRequiresIPhoneOS"))
 				Log.LogError (7021, info, MSBStrings.E7021, name);
 
 			var pluginsDir = Path.Combine (path, "PlugIns");
-			if (!Directory.Exists (pluginsDir)) {
+			if (!Directory.Exists (pluginsDir) && !isSingleProject) {
 				Log.LogError (7022, path, MSBStrings.E7022, name);
 				return;
 			}
 
-			int count = 0;
-			foreach (var plugin in Directory.EnumerateDirectories (pluginsDir, "*.appex")) {
-				ValidateWatchExtension (plugin, bundleIdentifier, shortVersionString, version);
-				count++;
-			}
+			if (!isSingleProject) {
+				int count = 0;
+				foreach (var plugin in Directory.EnumerateDirectories (pluginsDir, "*.appex")) {
+					ValidateWatchExtension (plugin, bundleIdentifier, shortVersionString, version);
+					count++;
+				}
 
-			if (count == 0)
-				Log.LogError (7022, pluginsDir, MSBStrings.E7022_A, name);
+				if (count == 0)
+					Log.LogError (7022, pluginsDir, MSBStrings.E7022_A, name);
+			}
 		}
 
 		void ValidateWatchExtension (string path, string watchAppBundleIdentifier, string mainShortVersionString, string mainVersion)

--- a/msbuild/Xamarin.iOS.Tasks/Tasks/ValidateAppBundleTaskBase.cs
+++ b/msbuild/Xamarin.iOS.Tasks/Tasks/ValidateAppBundleTaskBase.cs
@@ -115,7 +115,7 @@ namespace Xamarin.iOS.Tasks {
 				break;
 			}
 		}
-		
+
 		void ValidateWatchApp (string path, string mainBundleIdentifier, string mainShortVersionString, string mainVersion)
 		{
 			var name = Path.GetFileNameWithoutExtension (path);

